app/src/main/java/com/android/sample/ui/recipeOverview/RecipeOverview.kt

```diff
@@ -45,323 +39,454 @@ import androidx.compose.ui.graphics.Color
+        // - button
+        Button(
+            onClick = {
+              if (servingsCount > OVERVIEW_MIN_COUNTER_VALUE) onCounterChange(servingsCount - 1)
+            },
+            colors =
+                ButtonDefaults.buttonColors(
+                    containerColor = MaterialTheme.colorScheme.onBackground,
+                    contentColor = MaterialTheme.colorScheme.background),
+            modifier = Modifier.size(COUNTER_MIN_MAX_SIZE.dp).testTag("removeServings"),
+            contentPadding = PaddingValues(OVERVIEW_RECIPE_COUNTER_PADDING.dp)) {
+              Text(
+                  stringResource(R.string.counter_min),
+                  style = MaterialTheme.typography.titleMedium,
+                  color = MaterialTheme.colorScheme.background)
+            }
+
+        // Display the count
+        Text(
+            text = servingsCount.toString(),
+            style = MaterialTheme.typography.titleMedium,
+            color = MaterialTheme.colorScheme.background,
+            textAlign = TextAlign.Center,
+            modifier = Modifier.testTag("numberServings").width(OVERVIEW_COUNTER_TEXT_SIZE.dp))
+
+        // + button
+        Button(
+            onClick = {
+              if (servingsCount < OVERVIEW_MAX_COUNTER_VALUE) onCounterChange(servingsCount + 1)
+            },
+            colors =
+                ButtonDefaults.buttonColors(
+                    containerColor = MaterialTheme.colorScheme.onBackground,
+                    contentColor = MaterialTheme.colorScheme.background),
+            modifier = Modifier.size(COUNTER_MIN_MAX_SIZE.dp).testTag("addServings"),
+            contentPadding = PaddingValues(OVERVIEW_RECIPE_COUNTER_PADDING.dp)) {
               Text(
-                  servingsCountVar.toString(),
-                  fontSize = 12.sp,
-                  color = Color.Black,
-                  modifier = Modifier.testTag("numberServings"))
-              Button(
-                  onClick = { changeServingsCount(++servingsCountVar) },
-                  colors = ButtonColors(goldenBronze, Color.Black, goldenBronze, Color.Black),
-                  modifier = Modifier.size(20.dp).testTag("addServings"),
-                  contentPadding = PaddingValues()) {
-                    Text("+", fontSize = 12.sp)
-                  }
+                  stringResource(R.string.counter_max),
+                  style = MaterialTheme.typography.titleMedium,
+                  color = MaterialTheme.colorScheme.background)
             }
       }
-  Spacer(modifier = Modifier.height(7.dp))
-  IngredientsList(currentRecipe)
 }
 
-/** Display of the list of ingredients */
+/**
+ * Extracts the first integer from the measurement string, doubles it and replaces it in the string
+ *
+ * @param measurement: The measurement string
+ * @param servingsCount: The number of servings
+ * @return The modified string
+ */
+private fun extractAndDoubleFirstInt(measurement: String, servingsCount: Int): String {
```

Comment:
1. **Clear Logic**:
   - This method is straightforward and easy to follow—nice job keeping it readable!

2. **Rename for Clarity**:
   - Right now, `extractAndDoubleFirstInt` implies doubling, but you're actually multiplying by `servingsCount`. Maybe a name like `scaleFirstIntByServings` could clarify that it's based on the servings count.

3. **Error Handling**:
   - Consider wrapping the `toInt()` conversion with `runCatching` to handle any unexpected format issues without breaking the function.

   **Quick Example**:
   ```kotlin
   val originalNumber = runCatching { match.value.toInt() }.getOrElse { return measurement }
   ```

Reply:
1. Thank you!
2. Yes, this name is definitely an improvement.
3. I don’t think it’s necessary, as any issues would prevent the integer from being modified at all.